src/metpy/calc/thermo.py

```diff
@@ -362,8 +363,12 @@ def _lcl_iter(p, p0, w, t):
 
     # Conditional needed due to precision error with np.log in dewpoint.
     # Causes issues with parcel_profile_with_lcl if removed. Issue #1187
-    if np.isclose(lcl_p, pressure):
-        lcl_p = pressure
+    comp = np.isclose(lcl_p, pressure)
```

Comment:
I think this should work to eliminate the ugly conditional:
```python
lcl_p = np.where(np.isclose(lcl_p, pressure), pressure, lcl_p)
```

Reply:
Ah yeah that's much nicer. Test suite passes locally with that. Pushed up.